src/main/java/com/epam/aidial/cfg/dao/model/TimeTrackableEntity.java

```diff
@@ -0,0 +1,34 @@
+package com.epam.aidial.cfg.dao.model;
+
+import jakarta.persistence.MappedSuperclass;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.envers.Audited;
+
+import java.time.Instant;
+
+@Getter
+@Setter
+@Audited
+@MappedSuperclass
+public abstract class TimeTrackableEntity extends AbstractEntity<String> {
```

Comment:
let's left it generic as base class. don't see any reason to bind TimeTrackableEntity to AbstractEntity`<String>`.